docs/security.md

```diff
@@ -15,6 +15,24 @@
 
 # Security
 
+## Overview
+Trino-gateway has it's own security with it's own authetication and authorization.
+It's security is used only to authenticate/authorize it's own ui and the APIs.
+
+### Note for oauth2 flow:
+For oauth2 gateway uses `oidc/callback` where as trino uses `oauth2` path
+gateway should have it's own client id and all the trino clusters should have a single client id.
+gateway needs to pass thorugh the trino oauth2 requests to only one of the cluster. One way to handle that is to set a spectial rule like below:
```

Comment:
Should this be a bulleted list? Per our meeting Wednesday we should replace all instances of "gateway" with Trino Gateway.